addons/addon-base-raas/packages/base-raas-services/lib/aws-accounts/__tests__/aws-cfn-service.test.js

```diff
@@ -224,16 +263,17 @@ describe('AwsAccountService', () => {
       expect(res).toEqual('CURRENT');
     });
 
-    it('should correctly set account with undefined cfnStackName to NEEDSONBOARD', async () => {
-      // This account's status should change to 'NEEDSONBOARD'
+    it('should correctly set account with missing cfnStackName to ERRORED', async () => {
```

Comment:
uhhhhh....yes 😅
I could add a special case for this, but a missing stackname is equivalent to an invalid one, so in theory it should fail to find the stack name, and then return errored status.

Reply:
We need to avoid showing ERRORED status for pre-existing account, or customers would be filing bugs for it. A missing stackname can be differentiated from an invalid one right?

Reply:
yep, it can be--should I just default to UNKNOWN if there's no stack name?

Reply:
actually I'll probably default it to needs onboard

Reply:
added an additional check to set the status to UNKNOWN if stackname is either missing or empty. UNKNOWN state provides a button to deploy a new stack, and I just tested the flow and it seems to work pretty well.